design-docs/improved-console.md

```diff
@@ -0,0 +1,422 @@
+# Improved Console Output
+
+## Primary Goals
+* Improve perceived performance by making user aware of background work
+* Modern feel
+* User extensibility
+
+### User Goals for Console
+The user wants to answer the following questions:
+
+* When will my build finish?
+    * Soon (should I keep watching it) or later ("I'll check back")
+* Is there anything I should know about my build right now? (e.g. tests have failed, warnings)
+* What is Gradle doing right now?
+* Is my build successful?
+* What command did I execute?
+
```

Comment:
I think it is valuable to have a quick overview how much work was really done. We've seen on several occasions that users aren't even aware that their build is wrongly out of date because the many log messages make that hard to see. Now we are removing those log messages, so there is even less context to notice a problem. Just having a single percentage (work avoidance rate or something) could  help raise awareness. I realize build scans show all this and much more, but consider this: Many users aren't aware that their build has a problem and just assume that a 10s turnaround is normal, so the idea of creating a scan doesn't even occur to them. This leads to the wrong sentiment of "Gradle is slow", when really they would just need to fix their build.